### PR TITLE
[3.5] Backport: netutil: consistently format ipv6 addresses

### PR DESCRIPTION
**Pull request by Kidsan**
This is a backport of #15153

This formats ipv6 addresses to ensure they can be compared safely

Signed-off-by: kidsan <8798449+Kidsan@users.noreply.github.com>

### CODE REVIEW

**Review by ahrtr — APPROVED** [review]
LGTM

**Comment by ahrtr** [comment]
Please add a changelog item in https://github.com/etcd-io/etcd/blob/main/CHANGELOG/CHANGELOG-3.5.md

**Comment by Kidsan** [comment]
@ahrtr I hope I did that right: https://github.com/etcd-io/etcd/pull/15190

**Comment by ahrtr** [comment]
Leave it to @serathius to merge this PR since he has some comment in https://github.com/etcd-io/etcd/pull/15153#issuecomment-1406444646. 

Per my comment https://github.com/etcd-io/etcd/pull/15153#issuecomment-1406320600, I think it's OK to merge this PR for now and add whatever additional IPv6 related test case in separate PRs.

**Comment by serathius** [comment]
Just a note, we shouldn't do out of order backports. This fix is already in v3.4, forcing me to merge it as it would break clear git history. Backports for v3.4 should only take from PRs already backported to v3.5 and v3.5 should only take from fixes on main branch (assuming that issue is present on all releases). 

Next time let's agree on backport plan before we start merging.

**Comment by ahrtr** [comment]
> Backports for v3.4 should only take from PRs already backported to v3.5

I am afraid I can't agree this. 3.4 and 3.5 have huge differences, not every bug fix applies both branches. Backporting 3.4 should be independent with 3.5. The only agreement should be that both 3.4 and 3.5 should be backporting from main.

**Pull request merged**
